custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -219,6 +227,117 @@ def handle_exception(self, exception, repeat_record):
                                 "hiv_test_nikshay_registered", "hiv_test_nikshay_error")
 
 
+@RegisterGenerator(NikshayFollowupRepeater, 'case_json', 'JSON', is_default=True)
+class NikshayFollowupPayloadGenerator(BaseNikshayPayloadGenerator):
+    def get_payload(self, repeat_record, test_case):
+        self.test_case = test_case
+        occurence_case = get_occurrence_case_from_test(test_case.domain, test_case.get_id)
+        episode_case = get_open_episode_case_from_occurrence(test_case.domain, occurence_case.get_id)
+        person_case = get_person_case_from_occurrence(test_case.domain, occurence_case.get_id)
+
+        test_case_properties = test_case.dynamic_case_properties()
+        episode_case_properties = episode_case.dynamic_case_properties()
+
+        interval_id, lab_serial_number, result_grade, dmc_code = self._get_mandatory_fields(test_case_properties)
+
+        test_conducted_on = _format_date(test_case_properties, 'date_tested')
+        properties_dict = self._base_properties(repeat_record)
+        properties_dict.update({
+            "PatientID": episode_case_properties.get('nikshay_id'),
+            "TestDate": test_conducted_on,
+            "LabNo": lab_serial_number,
+            "Local_ID": person_case.get_id,
+            "IntervalId": interval_id,
+            # since weight is not taken and is mandatory we send 1
+            "PatientWeight": test_case_properties.get('weight', 1),
+            "SmearResult": result_grade,
+            "DMC": dmc_code
+        })
+
+        return json.dumps(properties_dict)
+
+    def _get_mandatory_fields(self, test_case_properties):
+        # list of fields that we want the case to have and should raise an exception if its missing or not in
+        # expected state to highlight missing essentials in repeat records. Check added here instead of
+        # allow_to_forward to bring to notice these records instead of silently ignoring them
+        interval_id = self._get_interval_id(test_case_properties.get('purpose_of_testing'),
+                                            test_case_properties.get('follow_up_test_reason'))
+
+        dmc_code = self._get_dmc_code(test_case_properties)
+        lab_serial_number = test_case_properties.get('lab_serial_number', None)
+        test_result_grade = test_case_properties.get('result_grade', None)
+        bacilli_count = test_case_properties.get('max_bacilli_count', None)
+        result_grade = self.get_result_grade(test_result_grade, bacilli_count)
+
+        if any(mandatory_value is None for mandatory_value in [lab_serial_number, result_grade]):
```

Comment:
Oh yes.... thanks for raising this Farid. Updating to check for blank string as well.